include/oneapi/dpl/functional

```diff
@@ -16,8 +16,20 @@
 #ifndef __ONEDPL_functional
 #define __ONEDPL_functional
 
-#include <functional>
+// Workarounds for libstdc++9, libstdc++10 when new TBB version is found in the environment
+#if __cplusplus >= 201703L
```

Comment:
It's not clear why we need to duplicate this workaround in `<functional>` header.

Reply:
What's the duplication?
If I remember correctly, we decided previously that the GCC workaround shall be applied to all oneDPL public headers.

Reply:
Hi, @akukanov 
"all oneDPL public headers" means all PSTL headers only or PSTL headers + tested API headers + random headers?
Currently, tested API and random headers don't include those GCC workaround.
Thanks very much.